engine/storage/src/main/java/org/apache/cloudstack/storage/image/deployasis/DeployAsIsHelperImpl.java

```diff
@@ -174,8 +174,26 @@ private void handleGuestOsFromOVFDescriptor(long templateId, String guestOsType,
                 hypervisor.toString(), minimumHypervisorVersion);
 
         if (CollectionUtils.isNotEmpty(guestOsMappings)) {
-            GuestOSHypervisorVO mapping = guestOsMappings.get(0);
-            long guestOsId = mapping.getGuestOsId();
+            Long guestOsId = null;
+            if (guestOsMappings.size() == 1) {
+                GuestOSHypervisorVO mapping = guestOsMappings.get(0);
+                guestOsId = mapping.getGuestOsId();
+            } else {
+                if (!StringUtils.isEmpty(guestOsDescription)) {
+                    int minimumEdits = Integer.MAX_VALUE;
+                    for (GuestOSHypervisorVO guestOSHypervisorVO : guestOsMappings) {
```

Comment:
some small worry here. I'm not sure it applies to guestos names, but linguistically this can yield false positives easily;
- "van os"
- "vanos"
- "fan nos"
- "vamos"
- "fam. os"
with the bare eye the last one is the proper replacement of the first. Would this algorithm yield this?